### DISCUSSION TITLE
Welcome to DRunge-kutta Discussions!

### DISCUSSION THREAD

**Discussion by fccoelho** [fccoelho/D-DifferentialEquations#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by baryluk** [comment]
Hi, I just spotted your post on dlang forums.

In 2004-2007 , I made a numerical library in D, heavily focused on ODEs (including symbolic integration, automatic differentiation, implicit methods, projection methods, RK methods, predictor-corrector methods, Richardson extrapolation methods, etc. etc.), with about 30 different ODE solvers. It was using templates and string mixing heavily (i.e. all RK methods are automatically generated from Butcher tables, and generate code with minimal amount of data copying or temporaries), to generate extremely optimized code, had lots of tests, etc.  After leaving university, I didn't have much use for it, so it kind of get abandoned, but the code is really nice, and I can dig it out, because it still is very useful, at least to look for. It was kind of a mess, because it was a port and extension of a library from 2002-2003, which was in pure C, and it was a mess. But D version was pretty good, it probably needs some updating to work with modern operator overloading in D. As a bonus it had (real and complex, double-double, quad-double + arbitrary precision) vectors, matrix, linear algebra, linear solvers, interval and affine arithmetic implementation, and dynamic parsing of expressions, to form mini computer algebra system, but mostly useful for differentiations.

**Reply by fccoelho** [reply]
Hi @baryluk , 

This is awesome! 
If you are willing to contribute this code, it would be a huge step towards having a usable D library for solving ordinary differential equations which is comparable in scope to those available to other languages. This has the potential to attract a lot of new users to D since, with a simple interface, Modellers out there can implement and simulate models just as easily as they currently do with Python or R, but have them run much faster. This is very relevant for people doing parameter estimation and uncertainty analysis through MCMC, which require solving the models thousands of times. 

So please feel free to create a branch on  this repo (or on your fork) to start adding you code. I'l be happy to Extend the Python validation script to compare the output of your solvers to that of Scipy solvers.
